cli/rack/__init__.py

```diff
@@ -350,12 +351,162 @@ def utility_copygraph_driver(base_url: Url, triple_store: Optional[Url], triple_
+        self,
+        from_path: Path,
+        to_path: Path,
+    ) -> None:
+        with open(from_path, mode='r', encoding='utf-8-sig') as f:
+            obj = yaml.safe_load(f)
+
+        frombase = from_path.parent
+        tobase = to_path.parent
+
+        files = obj['files']
+        for i in range(len(files)):
+            file = Path(files[i])
+            shutil.copyfile(frombase.joinpath(file), tobase.joinpath(file.name), follow_symlinks=True)
+            files[i] = file.name
+
+        c = obj.get('model-graphs')
+        if c is None:
+            self.model_graphs.add(str(MODEL_GRAPH))
+        elif type(c) == str:
+            self.model_graphs.add(c)
+        elif type(c) == list:
+            self.model_graphs.update(c)
+
+        with open(to_path, mode='w', encoding='utf-8-sig', newline='\n') as out: 
+            yaml.safe_dump(obj, out)
+
+    def data(
+        self,
+        from_path: Path,
+        to_path: Path,
+    ) -> None:
+        with open(from_path, mode='r', encoding='utf-8-sig') as f:
+            obj = yaml.safe_load(f)
+        frombase = from_path.parent
+        tobase = to_path.parent
+
+        for step in obj['ingestion-steps']:
+            if 'owl' in step:
+                path = Path(step['owl'])
+                shutil.copyfile(frombase.joinpath(path), tobase.joinpath(path.name), follow_symlinks=True)
+                step['owl'] = path.name
+            if 'csv' in step:
+                path = Path(step['csv'])
+                shutil.copyfile(frombase.joinpath(path), tobase.joinpath(path.name), follow_symlinks=True)
+                step['csv'] = path.name
+
+        self.data_graphs.add(obj['data-graph'])
+
+        c = obj.get('extra-data-graphs')
+        if c is not None:
+            self.data_graphs.update(c)
+
+        c = obj.get('model-graphs')
+        if c is not None:
+            if type(c) == str:
+                self.model_graphs.add(c)
+            elif type(c) == list:
+                self.model_graphs.update(c)
+
+        with open(to_path, mode='w', encoding='utf-8-sig', newline='\n') as out: 
+            yaml.safe_dump(obj, out)
+
+    def nodegroups(
+        self,
+        from_path: Path,
+        to_path: Path,
+    ) -> None:
+        shutil.copyfile(from_path.joinpath('store_data.csv'), to_path.joinpath('store_data.csv'), follow_symlinks=True)
+        with open(from_path.joinpath('store_data.csv'), 'r') as f:
+            for row in csv.DictReader(f):
+                json = row['jsonFile']
+                shutil.copyfile(from_path.joinpath(json), to_path.joinpath(json), follow_symlinks=True)
+
+    def manifest(
+        self,
+        from_path: Path,
+        to_path: Path,
+    ) -> None:
+
+        with open(from_path, mode='r', encoding='utf-8-sig') as f:
+            obj = yaml.safe_load(f)
+
+        base_path = from_path.parent
+        for i in range(len(obj['steps'])):
+            step = obj['steps'][i]
+            if 'manifest' in step:
```

Comment:
FYI, you can replace the above two lines with:

`for (i, step) in enumerate(obj['steps']):`